File: LAB1/Solucion(archivos)/primo_div_kprimo.c
#include <stdio.h>
#include <stdlib.h>

int num_es_primo(int num){
    int i;
    int cant_div=0;
    for(i=1;i<=num;i++){
        if(num%i==0)
            cant_div++;
    }
    //printf(" cant div = %d\n",cant_div);
    if(cant_div == 2)
        return 1;
    else
        return 0;
}

int num_divisible_k_primo(int num, int k){
    int i,j,cant_div_primo=0;
    for(i=1;i<=num;i++){
        if(num_es_primo(i)){
            if(num%i==0)
                cant_div_primo++;
        }
    }
    if(cant_div_primo == k)
        return 1;
    else    
        return 0;
}

int main(int argc, char const *argv[]){
    //Se piden 3 argumentos de entrada
    int a = atoi(argv[1]);
    int b = atoi(argv[2]);
    int c = atoi(argv[3]);
    //Numeros primos
    int num_en_rango=a;
    int num_primos_rango=0;
    int num_nums_div_k_primo=0;

    /*printf(" %d\n",a);
    printf(" %d\n",b);
    printf(" %d\n",c);*/

    //Numeros no primos divisibles por k-primos
    if (c==0){
        while(num_en_rango <= b){
            if(num_es_primo(num_en_rango))
                num_primos_rango++;
            num_en_rango++;
        }
        printf("Hay %d numeros primos", num_primos_rango);
    }
    else if(c>0){
        while(num_en_rango <= b){
            if(!num_es_primo(num_en_rango)){
                if(num_divisible_k_primo(num_en_rango,c))
                    num_nums_div_k_primo++;
            }
            num_en_rango++;
        }
        printf("Hay %d numeros divisibles por %d primo(s)", num_nums_div_k_primo, c);
    }
    
    return 0;
}

File: LAB1/Solucion(archivos)/primo_div_kprimo.py
import sys

#Funcion para saber si un numero es primo
def es_primo(numero):
    cant_div=0
    for i in range(1,numero+1): #Recorremos los valores desde 1 hasta numero
        if(numero%i==0): #Se evalua si es divisible por dicho valor a traves del operador modulo
            cant_div+=1 #Si es asi se incrementa la cantidad de divisores
    return cant_div == 2 #Retorna True si tiene dos divisores(primo) y False en caso contrario

#Funcion para obtener la cantidad de divisores primos de un numero
def div_por_kprimos(numero):
    cant_div_primos=0
    for i in range(1,numero+1):
        if(numero%i==0 and es_primo(i)): #Se evalua si es divisor y si es primo
            cant_div_primos+=1    
    return cant_div_primos #Retorna la cantidad de divisores primos
    
#Obtenemos los argumentos de entrada y los convertimos a enteros
a = int(sys.argv[1])
b = int(sys.argv[2])
c = int(sys.argv[3])
cant_primos = 0
cant_no_primos_div_por_kprimos=0

#Si el 3er argumento es 0 entonces contamos los primos en el rango [a-b]
if(c==0):
    for i in range(a,b+1):
        if(es_primo(i)):
            cant_primos+=1
    print("Hay {} numeros primo(s)".format(cant_primos))
#Caso contrario, debemos obtener los numeros no primos divisibles por la cantidad de primos igual a c  
elif(c>0):
    for i in range(a,b+1):
        if(not es_primo(i) and div_por_kprimos(i)==c): 
            cant_no_primos_div_por_kprimos+=1
    print("Hay {} numeros divisibles por {} primo(s)".format(cant_no_primos_div_por_kprimos, c))


